package.py

```diff
@@ -317,7 +317,10 @@ def close(self, failed=False):
         if failed:
             os.unlink(self._tmp_filename)
         else:
-            os.replace(self._tmp_filename, self.filename)
+            try:
```

Comment:
I guess a similar issue would happen when `failed` is `True`.

I'd rather check if `self._tmp_filename` exists before the `if failed:` above, and return early if it doesn't.

WDYT?